android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java

```diff
@@ -121,26 +130,20 @@ private void update(int coinType) {
             mKeyringService.getKeyringInfo(getSelectedCoinKeyringId(coinType),
                     keyringInfo -> { _mSelectedCoinKeyringInfoLiveData.postValue(keyringInfo); });
             mKeyringService.getKeyringsInfo(mSharedData.getEnabledKeyrings(), keyringInfos -> {
-                List<AccountInfo> accountInfos =
+                final List<AccountInfo> accountInfos =
                         WalletUtils.getAccountInfosFromKeyrings(keyringInfos);
                 _mAccountInfos.postValue(accountInfos);
                 _mKeyringInfosLiveData.postValue(keyringInfos);
-                mKeyringService.getSelectedAccount(coinType, accountAddress -> {
-                    if (accountAddress != null && !accountAddress.isEmpty()) {
-                        AccountInfo selectedAccountInfo = null;
-                        for (AccountInfo accountInfo : accountInfos) {
-                            if (accountInfo.address.equals(accountAddress)) {
-                                selectedAccountInfo = accountInfo;
-                                break;
-                            }
-                        }
-                        _mSelectedAccount.postValue(selectedAccountInfo);
-                    } else if (accountInfos.size() > 0) {
-                        AccountInfo accountInfo = accountInfos.get(0);
-                        _mSelectedAccount.postValue(accountInfo);
-                        setSelectedAccount(accountInfo.address, accountInfo.coin);
-                    }
-                });
+                if (coinType == CoinType.FIL) {
+                    mKeyringService.getFilecoinSelectedAccount(
+                            mSelectedFilecoinNetwork, accountAddress -> {
+                                updateSelectedAccountAndState(accountAddress, accountInfos);
+                            });
```

Comment:
`mSelectedFilecoinNetwork` will always be `FILECOIN_MAINNET` here (if isn't being updated via `addFilecoinAccount`).
Use [GetChainIdForActiveOrigin](https://github.com/brave/brave-core/blob/master/components/brave_wallet/common/brave_wallet.mojom#L1560) to get the selected network and use its `chainId`.

Reply:
No problem, I'll update these as well. Better to do it now.